### PR TITLE
Spark3: Add CREATE VIEW support

### LINKED ISSUE #1802: CREATE TEMPORARY VIEW not supported in Spark3 dialect
Following syntax is not supported in Spark3 dialect
`CREATE OR REPLACE TEMPORARY VIEW temp AS( SELECT * FROM temp2);`

## Expected Behaviour
Linting should be successful since it is [valid Spark3 syntax](https://spark.apache.org/docs/3.0.0/sql-ref-syntax-ddl-create-view.html#syntax)

## Observed Behaviour
Found unparsable section: 'CREATE OR REPLACE TEMPORARY VIEW temp AS (\n ...'

## Steps to Reproduce
- Try `sqlfluff lint` with above query

## Dialect
Spark3
## Version
sqlfluff, version 0.7.1

## Configuration
```
[tool.sqlfluff.core]
dialect = "spark3"


[tool.sqlfluff.templater.jinja]
apply_dbt_builtins = "False"

[tool.sqlfluff.templater.jinja.context.params]
anchor_date = "2020-01-01"
```

### PR DESCRIPTION
**Pull request by DipeshCS**
### Brief summary of the change made
fixes #1802

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_spark3.py [hunk 1]
```diff
@@ -113,9 +113,13 @@
         Ref("NotEqualToSegment_b"),
         Ref("LikeOperatorSegment"),
     ),
+    TemporaryGrammar=Sequence(
+        Sequence("GLOBAL", optional=True),
+        OneOf("TEMP", "TEMPORARY", optional=True),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_spark3.py:118]: I don't think this should be optional or you could have `GLOBAL` without `TEMP` or `TEMPORARY`.

```suggestion
        OneOf("TEMP", "TEMPORARY"),
```

Plus the whole block should be wrapped in an optional when called.

File: src/sqlfluff/dialects/dialect_spark3.py [hunk 2]
```diff
@@ -113,9 +113,13 @@
...
+        optional=True,
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_spark3.py:119]: I don't think you need this as it's optional when added:

```suggestion
```

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Some small tweaks but in general looks good.

Nice set of test cases!

File: src/sqlfluff/dialects/dialect_spark3.py (hunk 1 above)
Comment (DipeshCS) [reply src/sqlfluff/dialects/dialect_spark3.py]: Thanks for the review and suggestions!

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**